### ISSUE TITLE
v1.0.0-d25c1fd:beta - Snapshot height issue

### ISSUE THREAD

**Issue by alexdelprete** [jakowenko/double-take#101]
I'm trying to implement face recognition for my doorbell with Double Take. My doorbell has a 1536*2048 resolution, it's working fine with Frigate since 2 years.

Now, I needed to pass a full height snapshot to Double Take because of the face box size that was under minimum requirements with a height of 512. So I configured Double Take with this:

```
  attempts:
    # number of times double take will request a frigate latest.jpg for facial recognition
    latest: 0
    # number of times double take will request a frigate snapshot.jpg for facial recognition
    snapshot: 5
    # process frigate images from frigate/+/person/snapshot topics
    mqtt: true
    # add a delay expressed in seconds between each detection loop
    delay: 0

    image:
      # height of frigate image passed for facial recognition
      height: 2048
```

And Frigate like this:

```
    # Optional: Configuration for the jpg snapshots published via API
    snapshots:
      enabled: True
      clean_copy: False
      timestamp: False
      bounding_box: True
      crop: False
      height: 2048
      retain:
        default: 5 

    # Optional: Configuration for the jpg snapshots published via MQTT
    mqtt:
      # Optional: Enable publishing snapshot via mqtt for camera (default: shown below)
      # NOTE: Only applies to publishing image data to MQTT via 'frigate/<camera_name>/<object_name>/snapshot'.
      # All other messages will still be published.
      enabled: True
      # Optional: print a timestamp on the snapshots (default: shown below)
      timestamp: False
      # Optional: draw bounding box on the snapshots (default: shown below)
      bounding_box: False
      # Optional: crop the snapshot (default: shown below)
      crop: False
      # Optional: height to resize the snapshot to (default: shown below)
      height: 2048
      # Optional: jpeg encode quality (default: shown below)
      quality: 80
```

The problem is that the MQTT snapshot is correct, with a height of 2048, the snapshot is **always** an image with a 500 height no matter what value I use for image height in Double Take's config. I also checked under Frigate's clips folder, and found out that the images wrote there are indeed 2048 in height. Are those the actual ones passed via API to apps like Double Take?

I also had to disable latest.jpg API access because that image has not a fixed size, it's dynamic and I found no way to configure it in Frigate.

Thanks.

**Comment by alexdelprete** [comment]
When Double Take calls frigate's API for the snapshot, does it request a specific height, the one configured under the image: setting?

https://blakeblackshear.github.io/frigate/usage/api/#apieventsidsnapshotjpg

![image](https://user-images.githubusercontent.com/7027842/134002365-df9a0bc4-3c62-4016-b2c2-bd95dea895a5.png)

**Comment by jakowenko** [comment]
Hey @alexdelprete, when Frigate's API is used it does request a specific height. [Here](https://github.com/jakowenko/double-take/blob/beta/api/src/controllers/recognize.controller.js#L117) is the part of the code where that happens.

The snapshot.jpg and latest.jpg sizes are controlled from Double Take, but I think I have the indentation wrong in the beta docs currently.

Try nesting it like this.

```yaml
frigate:
  url: http://frigate:5000
  image:
    height: 2048
```

**Comment by alexdelprete** [comment]
Does FRIGATE.IMAGE.HEIGHT default to 512? That would explain it.

I've fixed indentation now, will let you know if it works now. I also noticed that snapshots are cropped while mqtt images are not.

**Comment by jakowenko** [comment]
> Does FRIGATE.IMAGE.HEIGHT default to 512? That would explain it.
> 
> I've fixed indentation now, will let you know if it works now. I also noticed that snapshots are cropped while mqtt images are not.

The default should be [`500`](https://github.com/jakowenko/double-take/blob/master/api/src/constants/defaults.js#L17).

Frigate allows cropping of the snapshot image via the API, so unless there's a mask configured DT requests a cropped image from Frigate. This allows recognition to happen quicker since the image is only of the object in question. If you think this should be a configurable field though, I can add support for that.

**Comment by alexdelprete** [comment]
> The default should be [`500`](https://github.com/jakowenko/double-take/blob/master/api/src/constants/defaults.js#L17).

So the indentation explains why it always defaulted to 500. Is there any validation of the configuration? Frigate for example doesn't start if there are indentation issues for parameters. Where can I check if there were config errors detected? Only in the container log? Would be nice to have logging in the UI. :)

**Comment by alexdelprete** [comment]
I stand corrected: the MQTT snapshots can't be controlled by DT obviously, so cropping has to be enabled in frigate. :)

Now everything's pretty clear.

**Comment by jakowenko** [comment]
The config is validated to make sure the YAML format is correct. If you try to enter a duplicate property for example you'll get an error. It doesn't validate though that a property you entered is in the correct location. I'd probably need to move to something like https://json-schema.org to support more advanced validation. Maybe I can include this in a future release.

![Screen Shot 2021-09-20 at 10 11 49 AM](https://user-images.githubusercontent.com/1081811/134017431-1a3395be-d1ec-44f5-ad09-6a0d1dfd9c49.png)

You can also copy the entire config using this button for easier debugging. This is also logged out when the container starts.

![Screen Shot 2021-09-20 at 10 12 16 AM](https://user-images.githubusercontent.com/1081811/134017579-a2a44596-24eb-4044-b56f-76aa7261d1aa.png)

**Comment by alexdelprete** [comment]
Indeed, config validation, particularly with yaml, is needed. That's one of the reasons I don't like yaml: one space a nothing works. :)

I noticed the copy config translated the config in json format. Can I then paste it again in that format or the parser expects only yaml?

**Comment by jakowenko** [comment]
> Indeed, config validation, particularly with yaml, is needed. That's one of the reasons I don't like yaml: one space a nothing works. :)
> 
> I noticed the copy config translated the config in json format. Can I then paste it again in that format or the parser expects only yaml?

Haha ya, I hear you about the YAML and the spacing. I still think it's easier for non-developers to work with that than JSON.

You can't currently copy the JSON format into the editor since it expects YAML. It's more used right now for debugging or a quick way to copy and paste your config into a Github issue. I can add a way to copy the YAML though if you'd like.

**Comment by alexdelprete** [comment]
Don't worry, I can copy&paste the yaml. :)

**Comment by jakowenko** [comment]
I'm going to close this since everything has been released in `1.0.0`. Feel fee to open a new issue if you run into problems.

**Comment by alexdelprete** [comment]
Sorry, I forgot to close it.
